### DISCUSSION TITLE
Variable for PublishPort in systemd .container

### DISCUSSION THREAD

**Discussion by LangDuTienTu** [containers/podman#24072] [Ideas]
Atm if I use variable for `PublishPort` such as `PublishPort=${PORT}:${PORT}`, quadlet dryrun would show error message and quadlet cannot generate systemd service

> invalid port format '${PORT}'

While we may use variable for `Volume`, why not `PublishPort` :3

**Comment by ygalblum** [comment]
Thanks for reporting this.

The reason it does not work for ports is that Quadlet does some validation checks: https://github.com/containers/podman/blob/main/pkg/systemd/quadlet/quadlet.go#L1957. But, I'm not sure it should still do them.

@vrothberg @Luap99 @rhatdan WDYT?

**Reply by vrothberg** [reply]
IMHO Quadlet should only validate/parse the input when there's a strong reason.  In this case, I think Quadlet could just pass on the value to --publish and let Podman take it from there.

**Reply by Luap99** [reply]
Yes agree, it should not parse/validate the values until it needs to (i.e. dependency resolution)

**Reply by LangDuTienTu** [reply]
Glad that you all agree with the idea, hope it would be in the next podman asap

**Reply by vrothberg** [reply]
@LangDuTienTu, would you open an issue for it?  That lets us better track the fix and users can subscribe to it.

**Reply by LangDuTienTu** [reply]
I just open an issue #24081 
This is my first time, so if I did anything wrong please pardon me :3

**Reply by vrothberg** [reply]
Thank you!

**Reply by rhatdan** [reply]
You can convert discussions to issues...
